src/app/destiny2/d2-definitions.service.ts

```diff
@@ -127,6 +132,9 @@ function getDefinitionsUncached(): IPromise<D2ManifestDefinitions> {
         defs[tableShort] = D2ManifestService.getAllRecords(db, table);
       });
 
+      const eagerTablesLoaded = 'eagerTablesLoaded';
+      defs[eagerTablesLoaded] = true;
```

Comment:
What's this for, and why can't it just be `defs.eagerTablesLoaded = true`?

Reply:
As part of this, you'll see that `header.tsx` does a polling loop on `getDefinitions`.

What I found was (and this doesn't make a whole lot of sense to me either), when you await definitions early enough, you get back a version of defs with the lazy loaded tables, but not the eager-loaded tables. Given that the underlying promise doesn't get returned until the eager tables are attached, this seems like something that shouldn't happen.

I put that in as a semaphore after the eager loading block to signal that the eager tables are really loaded in.

As far as the funky array assignment, `[ts] Property 'eagerTablesLoaded' does not exist on type '{}'.`

I was back in on the header (you mentioned item reviews could be pulling in D2 manifest for D1 users, and the same problem existed there) and waiting for the account to load seems to negate the race condition that existed here, so I dumped the new property.

Reply:
Hmm, that behavior you're describing is counterintuitive enough that I'd rather figure out exactly why you're seeing that, than work around it.